shared/android/c++/Binding.cpp

```diff
@@ -0,0 +1,140 @@
+#include "Binding.h"
+
+#include <dlfcn.h>
+// #include <forandroid.h>
+#include <android/log.h>
+#include <android/trace.h>
+#include <jsi/JSIDynamic.h>
+#include <chrono>  // std::chrono::seconds
+#include <thread>
+#if __i386__
+#include <keybaselib-386.h>
+#elif __arm__
+#include <keybaselib-arm.h>
+#else
+#include <keybaselib-arm64.h>
+#endif
+
+// #ifndef __ANDROID_API__
+// #define __ANDROID_API__ 29
+// #endif
+
+#define APPNAME "MyApp"
+
+struct EventHandlerWrapper {
+  EventHandlerWrapper(jsi::Function eventHandler)
+      : callback(std::move(eventHandler)) {}
+
+  jsi::Function callback;
+};
+
+#if ANDROID
+extern "C" {
+JNIEXPORT void JNICALL Java_io_keybase_ossifrage_MainActivity_install(
+    JNIEnv *env, jobject thiz, jlong runtimePtr) {
+  auto binding = std::make_shared<example::Binding>();
+  jsi::Runtime *runtime = (jsi::Runtime *)runtimePtr;
+
+  example::Binding::install(*runtime, binding);
+}
+}
+#endif
+
+namespace example {
+
+void Binding::install(jsi::Runtime &runtime, std::shared_ptr<Binding> binding) {
+  auto testModuleName = "nativeTest";
+  auto object = jsi::Object::createFromHostObject(runtime, binding);
+  runtime.global().setProperty(runtime, testModuleName, std::move(object));
+}
+
+Binding::Binding() {}
+
+jsi::Value Binding::get(jsi::Runtime &runtime, const jsi::PropNameID &name) {
+  auto methodName = name.utf8(runtime);
+
+  if (methodName == "traceBeginSection") {
+    return jsi::Function::createFromHostFunction(
+        runtime, name, 2,
+        [](jsi::Runtime &runtime, const jsi::Value &thisValue,
+           const jsi::Value *arguments, size_t count) -> jsi::Value {
+          auto name = arguments[0].toString(runtime).utf8(runtime);
+          ATrace_beginSection(name.c_str());
```

Comment:
Here we check that the method name is `traceBeginSection` (total call is `global.nativeTest.traceBeginSection`) and we return a function that will call the Android NDK's Atrace_beginSection. This is what lets us hook into Android's native systrace functionality.